sky/backends/cloud_vm_ray_backend.py

```diff
@@ -208,6 +208,9 @@ def add_prologue(self,
             self._code += [
                 textwrap.dedent(f"""\
                 _SETUP_CPUS = 0.0001
+                # Unset CUDA_VISIBLE_DEVICES (set by ray), since the setup command
+                # may need to detect the number of GPUs on the machine.
```

Comment:
```suggestion
                # The setup command will be run as a ray task with num_cpus=_SETUP_CPUS as the requirement; this means Ray will set CUDA_VISIBLE_DEVICES to an empty string. We unset it so that user setup command may properly use this env var.
```